viewing_party/party.py

```diff
@@ -0,0 +1,227 @@
+
+    return unique_movies
+
+
+def get_friends_unique_watched(user_data):
+
+    # collect the title of movies I have watched
+    my_watched_movie_titles = set()
+    for movie in user_data["watched"]:
+        my_watched_movie_titles.add(movie["title"])
+
+    # go through my friends movies in an output list that are not movies in my watched list
+    friends_unique_movies = {}
+    for friends_data in user_data["friends"]:
+        for movie in friends_data["watched"]:
+            if movie["title"] not in my_watched_movie_titles:
+                friends_unique_movies[movie["title"]] = movie
+
+    return friends_unique_movies.values()
+
+
+# ************   Wave 4   ************
+
+# user_data = {
+#    "supscriptions": [<"Service A">] - streaming services that user has access
+#    "watched": [],
+#    "friends": [
+#            {watched: [{"title": <"Title A">, "host": <"Service A"},
+#            {watched: [{"title": <"Title B">, "host": <"Service B"} ]
+# }
+
+
+def get_available_recs(user_data):
+    # Make a list of the titles of movies I've watched
+    watched_titles = []
+    for movie in user_data["watched"]:
+        watched_titles.append(movie["title"])
+
+    # Now gather friend's movies that:
+    # - Are available on my servies
+    # - that I haven't watched
+    recommended_movies = []
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie["host"] in user_data["subscriptions"] and movie["title"] not in watched_titles:
+                recommended_movies.append(movie)
+                watched_titles.append(movie["title"])
+
+    return recommended_movies
+
+# ************   Wave 5   ************
+
+
+"""
+user_data= {
+        "favorites": [],
+        "watched": [
+            {"title": "Title A",
+            "genre": "Intrigue"
+            },
+            {"title": "Title B",
+            "genre": "Intrigue"
+            }],
+        "friends": [
+            {"watched": [
+                    {"title": "Title D",
+                    "genre": "Intrigue"}]
+            },
+            {"watched": [
+                    {"title": "Title C",
+                    "genre": "Fantasy"
+                    }]
+            }
+        ]
+    }
+
+"""
+
+# - add to list if user hasn't watched the movies
+# - at least one friend has watched it.
+# - most frequent genre
+
+
+def get_new_rec_by_genre(user_data):
+    # calling most watched genre function
+    favorite_genre = get_most_watched_genre(user_data)
+
+    watched_titles = []
+    for movie in user_data["watched"]:
+        watched_titles.append(movie["title"])
+
+    recommended_movies = []
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie["genre"] == favorite_genre and movie["title"] not in watched_titles:
+                recommended_movies.append(movie)
+                watched_titles.append(movie["title"])
+
+    return recommended_movies
+
+
```

Comment:
💡 Quick Tip: To make your project neater and follow [pep8 guidelines](https://www.python.org/dev/peps/pep-0008/), you can have VS Code autoformat your files by right-clicking on a file in VS Code and select Format Document With... then select python from the drop-down menu. You'll then be prompted to install autopep8, click yes and the package will be installed within your virtual environment.